### PR TITLE
chore(ui-responsive): add TS types to `ui-responsive` package

### PR DESCRIPTION
**Pull request by ke1k0**
Closes: INSTUI-3077

### CODE REVIEW

File: packages/ui-responsive/src/addElementQueryMatchListener.ts [hunk 1]
```diff
@@ -76,27 +80,35 @@ import { debounce } from '@instructure/debounce'
...
+  query: BreakpointQueries,
+  el:
+    | Node
+    | Window
+    | React.ReactElement
+    | React.Component
+    | ((
+        ...args: any[]
+      ) => Node | Window | React.ReactElement | React.Component),
+  cb: (queriesMatching: QueriesMatching) => any
```
Comment (matyasf) [inline packages/ui-responsive/src/addElementQueryMatchListener.ts:98]: `cb` is `UpdateMatches` type (from index.tsx)

The signature of this method should be the same as `addMediaQueryMatchListener`, I'd extract it to a type.

Comment (matyasf) [reply packages/ui-responsive/src/addElementQueryMatchListener.ts:98]: otherwise it looks good!

Comment (ke1k0) [reply packages/ui-responsive/src/addElementQueryMatchListener.ts:98]: `addMediaQueryMatchListener` sadly has an extra parameter `matchMedia` :S I couldn't find a way to extract it in a way so one can have an extra parameter and the other doesn't.

**Review by Brailor — APPROVED** [review]

**Review by matyasf — APPROVED** [review]
looks good!

**Pull request merged**